samples/dfu/nrf5340/boards/CMakeLists.txt

```diff
@@ -0,0 +1,9 @@
+#
+# Copyright (c) 2022 INTELLINIUM <giuliano.franchetto@intellinium.com>
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/${CONFIG_BOARD}.c")
+    target_sources(app PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/${CONFIG_BOARD}.c)
```

Comment:
This doesn't make sense, you are using your custom board in a repository somewhere, correct? Your board files should not be here, they are also incomplete.

Reply:
No, the idea here is to have specific code needed per board for the sample to work. This is not to include the board-related source files that can be found in boards/arm/...
The build system can find and use ${BOARD}.conf and ${BOARD}.overlay placed in the boards/ directory of any application, but I also need a ${BOARD}.c file to be built for some initialization.
My board needed some special init code for this sample to work.
I'll remove it, I just needed it to simply switch between board without changing the code of the sample